packages/twenty-front/src/pages/settings/workspace/hooks/useCheckCustomDomainValidRecords.ts

```diff
@@ -0,0 +1,32 @@
+import { customDomainRecordsState } from '~/pages/settings/workspace/states/customDomainRecordsState';
+import { useCheckCustomDomainValidRecordsMutation } from '~/generated/graphql';
+import { isDefined } from 'twenty-shared';
+import { useSetRecoilState } from 'recoil';
+
+export const useCheckCustomDomainValidRecords = () => {
+  const [checkCustomDomainValidRecords] =
+    useCheckCustomDomainValidRecordsMutation();
+
+  const setCustomDomainRecords = useSetRecoilState(customDomainRecordsState);
+
+  const checkCustomDomainRecords = () => {
+    setCustomDomainRecords((currentState) => ({
+      ...currentState,
+      loading: true,
+    }));
+    checkCustomDomainValidRecords({
+      onCompleted: (data) => {
```

Comment:
onCompleted won't be executed on sub sequent calls, it is only executed on the first request, are you sure that's what you want ? If not it would be better to use a useEffect with a boolean state to know if that's already done.

Reply:
Why won't it be executed on subsequent calls?
It is executed when the page is loaded and when the user clicks the refresh button.

Reply:
https://github.com/apollographql/react-apollo/issues/3709